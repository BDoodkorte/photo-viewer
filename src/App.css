@import "./PhotoSelector/PhotoSelector.css";

#root {
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

* {
  box-sizing: border-box;
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}

@media only screen and (min-width: 1300px) {

  .mainpiccontainer {
    max-width: 100%;
  
  }

}

@media only screen and (max-width: 1299px) {
  
  
  .mainpiccontainer {
    max-width: 100%;
  
  }

}

@media only screen and (max-width: 650px) {

  .mainpiccontainer {
    max-width: 100%;
  
  }

}
