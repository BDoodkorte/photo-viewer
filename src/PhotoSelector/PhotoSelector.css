@media only screen and (min-width: 1300px) {

    .photogrid {
        display: inline-flex;
        flex-wrap: wrap;
        gap: 0.5%;
        row-gap: 7px;
        min-width: 1280px;
      }

    .thumbnails {
      width: 24.625%;
    }
    
    .border {
      border: solid 4px rgb(255, 196, 0);
      width: 24.625%;
    }

  }
  
  @media only screen and (max-width: 1299px) {
    .photogrid {
        display: inline-flex;
        flex-wrap: wrap;
        gap: 1%;
        row-gap: 7px;
        min-width: 325px;
      }

    .thumbnails {
      width: 32%;
    }
    
    .border {
      border: solid 4px rgb(255, 196, 0);
      width: 32%;
    }

  }
  
  @media only screen and (max-width: 650px) {
    .photogrid {
        display: inline-flex;
        flex-wrap: wrap;
        gap: 1%;
        row-gap: 5px;
        min-width: 325px;
      }
    .thumbnails {
      width: 49%;
    }
    
    .border {
      border: solid 4px rgb(255, 196, 0);
      width: 49%;
    }
  }