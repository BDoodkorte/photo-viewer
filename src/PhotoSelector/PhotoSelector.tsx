import React from "react";  // import React (to provide access to TSX)

// declare and export new function called 'PhotoSelector'
export const brokenImages = [
	1, 24, 32, 36, 44, 47
];

function PhotoSelector() {
	const urls = [];

	for (let i = 0; i < 50; i++) {
		if (!brokenImages.includes(i)) {
			const imageNumberString = i.toString().padStart(2, '0');
			urls.push(`https://picsum.photos/id/6${imageNumberString}/1600/900.jpg`)
		}
	}

	return urls;
}

export const photoSelection = PhotoSelector();


