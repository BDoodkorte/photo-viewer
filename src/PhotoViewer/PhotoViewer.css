.main_pic {
max-width: 1280px;
}