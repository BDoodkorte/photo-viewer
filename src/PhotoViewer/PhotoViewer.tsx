import React from "react";  // import React (to provide access to TSX)

export function PhotoViewer(props) {    // declare and export new function called 'PhotoViewer'
    return (                
        <div className="photo">    
            <img src={props.url} /> 
        </div>
    );
}
