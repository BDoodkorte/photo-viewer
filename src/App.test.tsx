import React from 'react';
import * as renderer from 'react-test-renderer';
import { getByAltText, render } from '@testing-library/react';
import App from './App';
import { brokenImages, photoSelection } from './PhotoSelector/PhotoSelector';
import { PhotoViewer } from './PhotoViewer/PhotoViewer';
import { fireEvent } from '@testing-library/react';




// Test test to check if the word photo appears on the webpage
test('renders Photo text', () => {
    const { getByText } = render(<App />);
    const textElement = getByText(/Photo/i);
    expect(textElement).toBeInTheDocument();
});


//A unit test to check our imageUrl generation code - 
// for me, this might check that the first link is what I expect and that it doesn’t include the ‘broken’ images.
test('checks image list does not contain broken images', () => {
    const brokenUrls: any = [];
    for (let i = 0; i < brokenImages.length; i++) {
        brokenUrls.push(`https://picsum.photos/id/6${brokenImages[i]}/1600/900.jpg`)
    }
    function findCommonElements3(arr1: any[], arr2: string | any[]) {
        return arr1.some(item => arr2.includes(item))
    }
    let check = findCommonElements3(photoSelection, brokenUrls)
    expect(check).toBeFalsy();
});


// A ‘Regression / Snapshot' test to confirm that I the ‘ImageViewer' component doesn’t accidentally change in the future.
describe("My Component", () => {
  it("Should match snapshot with url prop", async () => {
      const tree = renderer.create(<PhotoViewer url="https://picsum.photos/id/600/1600/900"/>).toJSON();
      expect(tree).toMatchSnapshot();
    });
});


// A Component Test to confirm that when I click a thumbnail, 
// then the thumbnail becomes selected and the ImageViewer updates with the new image.
test('does image update when clicked', () => {

    const loadApp = render(<App />);
    // const that stores selected thumbnail url
    const thumbnailTest : any = loadApp.getByAltText("https://picsum.photos/id/639/1600/900.jpg");

    // const that stores mainImage url
    const mainImageTest : any = loadApp.getByAltText("main_pic");
    expect(mainImageTest).toHaveAttribute("src","https://picsum.photos/id/600/1600/900");

    
    fireEvent.click(thumbnailTest); //this needs to be the element holding the url, cant run fireevent on url itself
    const mainImageTest2 : any = loadApp.getByAltText("main_pic");


    expect(mainImageTest2).toHaveAttribute("src","https://picsum.photos/id/639/1600/900.jpg");
});