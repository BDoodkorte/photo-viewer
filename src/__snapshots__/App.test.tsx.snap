// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Component Should match snapshot with url prop 1`] = `
<div
  className="photo"
>
  <img
    alt="main_pic"
    className="main_pic"
    src="https://picsum.photos/id/600/1600/900"
  />
</div>
`;
